### PR TITLE
Add initial notifier support.

### PR DESCRIPTION
**Pull request by mrstegeman**
References #1758
Requires mozilla-iot/gateway-addon-node#39

### CODE REVIEW

**Comment by mrstegeman** [comment]
This requires that the gateway-addon changes be pulled in manually.

Also, there are no tests yet. Wanted to make sure the API was ok before doing those.

File: src/addon-manager.js [hunk 1]
```diff
@@ -828,13 +971,22 @@ class AddonManager extends EventEmitter {
...
+    const notifierIds = notifiers.map((n) => n.id);
+
     const unloadPromises = [];
     if (adapters.length > 0) {
       for (const a of adapters) {
         console.log('Unloading', a.name);
         unloadPromises.push(a.unload());
         this.adapters.delete(a.id);
       }
+    } else if (notifiers.length > 0) {
```
Comment (hobinjk) [inline src/addon-manager.js]: I think a single package could expose both an add-on and a notifier if it wanted to

Comment (mrstegeman) [reply src/addon-manager.js]: Yes, it could. I was considering doing that for the email adapter for backwards compat.

**Comment by mrstegeman** [comment]
The rules test failure is occurring on master as well, so it's not due to this PR.

**Pull request merged**